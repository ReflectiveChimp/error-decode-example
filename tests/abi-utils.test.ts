import { expect, test } from 'bun:test';
import { extractErrors, readAbi } from '../src/lib/abi-utils.js';

const abi = await readAbi(__dirname + '/abi.json');

test('extractErrors', async () => {
  expect(extractErrors(abi)).toMatchSnapshot();
});