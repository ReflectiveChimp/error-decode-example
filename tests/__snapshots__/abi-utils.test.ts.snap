// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`extractErrors 1`] = `
[
  {
    "selector": "0xcc35f152",
    "signature": "NoSwapData(address,address)",
  },
  {
    "selector": "0x43038e14",
    "signature": "PriceFailed(address)",
  },
  {
    "selector": "0x71c4efed",
    "signature": "SlippageExceeded(uint256,uint256)",
  },
  {
    "selector": "0x6f40b568",
    "signature": "SwapFailed(address,bytes)",
  },
]
`;
